api/tests/src/api/applications/test_application_routes.py

```diff
@@ -98,3 +102,306 @@ def test_application_start_invalid_request(
     # Verify no application was created
     applications_count = db_session.execute(select(Application)).scalars().all()
     assert len(applications_count) == 0
+
+
+def test_application_form_update_success_create(
+    client, enable_factory_create, db_session, api_auth_token
+):
+    """Test successful creation of an application form response"""
+    opportunity = OpportunityFactory.create()
+    competition = CompetitionFactory.create(opportunity_id=opportunity.opportunity_id)
+
+    # Create application
+    application = Application(
+        application_id=uuid.uuid4(), competition_id=competition.competition_id
+    )
+    db_session.add(application)
+
+    # Create form
+    form = FormFactory.create(
+        form_json_schema={"type": "object", "properties": {"name": {"type": "string"}}},
+        form_ui_schema={"name": {"ui:widget": "text"}},
+    )
+    db_session.add(form)
+    db_session.commit()
```

Comment:
The create function for a factory already does add/commit in it (which is why a lot of create calls is slow as it's a lot of separate commits).

```suggestion
    # Create form
    form = FormFactory.create(
        form_json_schema={"type": "object", "properties": {"name": {"type": "string"}}},
        form_ui_schema={"name": {"ui:widget": "text"}},
    )
```